### PR TITLE
Finishing

### PR DESCRIPTION
**Pull request by Shilpi75**
Distance utils added
Code adjusted to use distance units

### CODE REVIEW

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/DistanceUtils.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+
+import app.tasknearby.yashcreations.com.tasknearby.R;
+
+/**
+ * Lists all distance utility functions.
+ *
+ * @author
```
Comment (vermayash8) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/DistanceUtils.java]: Don't leave it blank

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/DistanceUtils.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+public class DistanceUtils {
+
+    /**
+     * Returns formatted distance string in metric/imperial units according to user's settings.
+     */
+    public static String getFormattedDistanceString(Context context, double distance) {
+
+        // Check for units in settings.
+        SharedPreferences defaultPref = PreferenceManager.getDefaultSharedPreferences(context);
+        String unitsPref = defaultPref.getString(context.getString(R.string.pref_unit_key),
```
Comment (vermayash8) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/DistanceUtils.java]: On checking I found that the key being used currently `pref_unit_key` is `unit` which is not consistent with the old one `units`. Change it to `units` . Please check for the rest too.
[Reference link](https://github.com/vermayash8/TaskNearby/blob/4878def3111e4471e138d281f3728f8d12f87034/app/src/main/res/values/strings.xml#L45)

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/DistanceUtils.java (hunk 1 above)
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/DistanceUtils.java]: Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/DistanceUtils.java (hunk 2 above)
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/DistanceUtils.java]: Okay

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java [hunk 3]
```diff
@@ -89,4 +92,23 @@ public void onBackPressed() {
...
+        if (!defaultPref.contains(getString(R.string.pref_power_saver_key))) {
+            // It means user has updated the app and opening this version for the first time.
+            String accuracy = defaultPref.getString(getString(R.string.pref_accuracy_key),
+                    getString(R.string.pref_accuracy_default));
+            if (accuracy.equals(getString(R.string.pref_accuracy_balanced))) {
+                // Set power saver mode.
+                SharedPreferences.Editor editor = defaultPref.edit();
+                editor.putBoolean(getString(R.string.pref_power_saver_key), true);
+                editor.apply();
+            }
```
Comment (vermayash8) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java]: Consider a scenario when the `pref_accuray = high`. Now when that user opens the app for the first time, `pref_power_saver` is not set anywhere. So, what happens is, when he'll open the app for the next time, then all these checks will go again till line 106. 
So, also set this `pref_power_saver = false` even in the high case.

Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java]: Yes, right. Done

File: app/src/main/res/layout/activity_about.xml [hunk 4]
```diff
@@ -32,6 +32,7 @@
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         android:layout_marginEnd="8dp"
+        android:layout_marginStart="2dp"
```
Comment (Shilpi75) [inline app/src/main/res/layout/activity_about.xml]: This is done to have spacing between 'Task Nearby' and 'v4.0' in About activity.

**Pull request merged**